Week2/app1.js

```diff
@@ -0,0 +1,137 @@
+"use strict";
+{
+    let fetchJSON = (URL)=>{
+    return new Promise((resolve, reject)=>{
+        const dataRepositories = new XMLHttpRequest;
+        dataRepositories.responseType='json';
+        dataRepositories.open ('GET', URL);
+
+            dataRepositories.onload= ()=> {
+                if (dataRepositories.status<400){ 
+                    resolve(dataRepositories.response);
+                }else {
+                    reject(new Error(`Network error: ${dataRepositories.status} - ${dataRepositories.statusText}`));
+                }
+            };
+
+        dataRepositories.onerror= ()=>{
+            reject(new Error('Network request failed'));
+        }
+        dataRepositories.send();
+    })
+    
+    };
+
+    function createAppend (name, parent, options = {}){
+        const elem = document.createElement(name);
+        parent.appendChild(elem);
+        Object.keys(options).forEach(key =>{
+            if (key === "text"){
+                const optionsKey=options[key].replace(/['"]+/g, '');
+
+                elem.innerHTML= optionsKey;
+
+            }else{
+                elem.setAttribute(key, options[key]);
+            };
+        });
+              
+        return elem;
+    }
+
+    const URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+    
+    function renderRepository(container, repository){
+        
+        container.innerHTML='';
+        
+        const table1 = createAppend('tb',container,{id:"table1"});
+        const tBody1 = createAppend('tbody',table1);
+        const tr = createAppend('tr',tBody1);
+        const tr1 = createAppend('tr',tBody1);
+        const tr2 = createAppend('tr',tBody1);
+        const tr3 = createAppend('tr',tBody1);
+
+        createAppend('td', tr, { text:"Repository: ", class:"tableData" });
+
+        const nameTag= createAppend('a',tr, {text: JSON.stringify(repository.name)});
+        nameTag.setAttribute('href', repository.html_url);
+        nameTag.setAttribute('target', '_blank');
+
+        createAppend('td', tr1, { text:"Description: ", class:"tableData" });
+        createAppend('td',tr1, {text: JSON.stringify(repository.description)});
+
+        createAppend('td', tr2, { text:"Fork: ", class:"tableData" });
+        createAppend('td',tr2, {text: JSON.stringify(repository.forks_count)});
+        
+        createAppend('td', tr3, { text:"Updated: ", class:"tableData" });
+        let updateDate = new Date (repository.updated_at);
+        createAppend('td',tr3, {text: updateDate.toLocaleString()});
+
+        
+
+    }
```

Comment:
The preceding three blank lines are meaningless. Use blank lines conciously to separate blocks of related code like you would use blanks lines to separate paragraphs in a piece of written text. Do not use more than one blank linen at a time. Use a single blank lines between function definitions. Do no use randomly insert blank lines.